opensfm/src/map/test/map_test.cc

```diff
@@ -0,0 +1,147 @@
+#include <map/map.h>
+#include <geometry/camera.h>
+#include <sfm/observation.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+
+namespace {
+// TEST(Map, CreateAndDeleteShots) {}
+
+TEST(Map, CreateCamera) {
+  Camera camera = Camera::CreatePerspectiveCamera(0.5, 0, 0);
+  camera.width = 640;
+  camera.height = 480;
+  auto map = map::Map();
+  constexpr auto n_cameras{10};
+  for (auto i = 0; i < n_cameras; ++i)
+  {
+    camera.id = std::to_string(i);
+    auto* my_cam = map.CreateCamera(camera);
+    ASSERT_EQ(my_cam->id, camera.id);
+    ASSERT_EQ(my_cam->width, camera.width);
+    ASSERT_EQ(my_cam->height, camera.height);
+  }
+  ASSERT_EQ(map.NumberOfCameras(), n_cameras);
+  ASSERT_EQ(map.GetAllCameras().size(), n_cameras);
+}
+
+TEST(Map, CreateAndDeleteShots) 
+{
+  Camera camera = Camera::CreatePerspectiveCamera(0.5, 0, 0);
+  camera.width = 640;
+  camera.height = 480;
+  camera.id = "cam1";
+  Camera camera2 = Camera::CreatePerspectiveCamera(0.5, 0, 0);
+  camera2.width = 640;
+  camera2.height = 480;
+  camera2.id = "cam2";
+
+  auto map = map::Map();
+  constexpr auto n_shots{20};
+  auto* my_cam = map.CreateCamera(camera);
+  map.CreateCamera(camera2);
+  for (auto i = 0; i < n_shots; ++i)
+  {
+    const auto shot_id = "shot"+std::to_string(i);
+    auto* shot = map.CreateShot(shot_id, i < n_shots/2 ? camera.id : camera2.id);
+    ASSERT_EQ(shot, map.GetShot(shot_id));
+    ASSERT_EQ(shot->id_, shot_id);
+    ASSERT_EQ(shot->unique_id_, i);
+    ASSERT_TRUE(map.HasShot(shot_id));
```

Comment:
Would it be possible to have a fixture and split this test so we have separate tests for creation and deletion ?